src/axom/primal/geometry/NURBSCurve.hpp

```diff
@@ -0,0 +1,1177 @@
+    {
+      newControlPoints[i + r] = m_controlPoints[i];
+      if(isRational)
+      {
+        newWeights[i + r] = m_weights[i];
+      }
+    }
+
+    // Insert the new control points
+    CoordsVec tempControlPoints(p + 1);
+    WeightsVec tempWeights(isRational ? p + 1 : 0);
+    for(int i = 0; i <= p - s; ++i)
+    {
+      for(int N = 0; N < NDIMS; ++N)
+      {
+        tempControlPoints[i][N] = m_controlPoints[span - p + i][N] *
+          (isRational ? m_weights[span - p + i] : 1.0);
+      }
+
+      if(isRational)
+      {
+        tempWeights[i] = m_weights[span - p + i];
+      }
+    }
+
+    // Insert the knot r times
+    axom::IndexType L;
+    for(int j = 1; j <= r; ++j)
+    {
+      L = span - p + j;
+      for(int i = 0; i <= p - j - s; ++i)
+      {
+        T alpha =
+          (t - m_knotvec[L + i]) / (m_knotvec[i + span + 1] - m_knotvec[L + i]);
+
+        tempControlPoints[i].array() =
+          (1.0 - alpha) * tempControlPoints[i].array() +
+          alpha * tempControlPoints[i + 1].array();
+
+        if(isRational)
+        {
+          tempWeights[i] =
+            (1.0 - alpha) * tempWeights[i] + alpha * tempWeights[i + 1];
+        }
+      }
+
+      for(int N = 0; N < NDIMS; ++N)
+      {
+        newControlPoints[L][N] =
+          tempControlPoints[0][N] / (isRational ? tempWeights[0] : 1.0);
+        newControlPoints[span + r - j - s][N] = tempControlPoints[p - j - s][N] /
+          (isRational ? tempWeights[p - j - s] : 1.0);
+      }
+
+      if(isRational)
+      {
+        newWeights[L] = tempWeights[0];
+        newWeights[span + r - j - s] = tempWeights[p - j - s];
+      }
+    }
+
+    for(auto i = L + 1; i < span - s; ++i)
+    {
+      for(int N = 0; N < NDIMS; ++N)
+      {
+        newControlPoints[i][N] =
+          tempControlPoints[i - L][N] / (isRational ? tempWeights[i - L] : 1.0);
+      }
+
+      if(isRational)
+      {
+        newWeights[i] = tempWeights[i - L];
+      }
+    }
+
+    // Update the knot vector and control points
+    m_knotvec.insertKnot(span, t, r);
+    m_controlPoints = newControlPoints;
+    m_weights = newWeights;
+
+    return span + r;
+  }
+
+  /*!
+   * \brief Splits a NURBS curve into two curves at a given parameter value
+   *
+   * \param [in] t parameter value between 0 and 1 at which to evaluate
+   * \param [out] n1 First output NURBS curve
+   * \param [out] n2 Second output NURBS curve
+   * \param [in] normalize Whether to normalize the output curves
+   * 
+   * \pre Parameter \a t must be in the span of the knots
+   */
+  void split(T t,
+             NURBSCurve<T, NDIMS>& n1,
+             NURBSCurve<T, NDIMS>& n2,
+             bool normalize = false) const
+  {
+    SLIC_ASSERT(t >= m_knotvec[0] && t <= m_knotvec[m_knotvec.getNumKnots() - 1]);
+
+    const bool isCurveRational = this->isRational();
+    const int p = getDegree();
+
+    // Handle the special case of splgitting at the endpoints
+    if(t == 0.0)
```

Comment:
Should this be `0` or the lowest knot vector value?
Similarly for the check in the `else if t == 1.0` branch

Reply:
Whoops! Should totally be the min/max knot value.